### ISSUE TITLE
Switch to setuptools, remove distutils

### ISSUE THREAD

**Issue by tito** [kivy/kivy#5275]
Distutils is still the main library used for our packaging.

I would like to propose switching now by default on setuptools. The previous issues with distutils and Cython integration on old debian could be just now unsupported (i don't remember exactly why, but we may recover the why in the commit history / issues / ml.)

The reason is today, i installed Kivy using "pip -e ." from my Kivy directory. When uninstalling, pip said:
```
    DEPRECATION: Uninstalling a distutils installed project (Kivy) has been deprecated and will be removed in a future version. This is due to the fact that uninstalling a distutils project will only partially uninstall the project.
```

So it's time to fully move.

**Comment by KeyWeeUsr** [comment]
Are we going to remove distutils completely later, or is it only about the default tool? Also, I'm not sure, but weren't there some problems with setuptools and android/p4a?

**Comment by matham** [comment]
We need setuptools to build wheels. However, setuptools apparently doesn't
work on android.

**Comment by misl6** [comment]
Fixed via #8070
